lib/src/functions/map.dart

```diff
@@ -97,23 +98,23 @@ final _hasKey = _function("has-key", r"$map, $key", (arguments) {
   return SassBoolean(map.contents.containsKey(key));
 });
 
-SassMap _set(SassMap map, List<Value> keys, Value value) {
+/// Updates a map with the given [value].
+///
+/// If more than one key is provided, this means the map targeted for update is
+/// nested within [map]. The multiple [keys] form a path of nested maps that
+/// leads to the targetted map. If any value along the path is not a map, this
+/// creates and inserts a new map at that key.
+SassMap _setImpl(SassMap map, List<Value> keys, Value value, [int index = 0]) {
   var mutableMap = Map.of(map.contents);
-  var key = keys.first;
+  var key = keys[index];
 
-  if (keys.length == 1) {
+  if (index == keys.length - 1) {
     mutableMap[key] = value;
     return SassMap(mutableMap);
   }
 
-  SassMap nestedMap;
-  try {
-    nestedMap = mutableMap[key].assertMap();
-  } on SassScriptException {
-    nestedMap = SassMap({});
-  }
-
-  mutableMap[key] = _set(nestedMap, keys.sublist(1), value);
+  var nestedMap = mutableMap[key].tryMap() ?? SassMap({});
```

Comment:
```suggestion
  var nestedMap = mutableMap[key].tryMap() ?? const SassMap.empty();
```

Using `const` constructors ensures that the same empty map instance is re-used everywhere, which saves allocations.

Reply:
Don't we want new empty map instances to be created for each key?

Reply:
SassScript values are all immutable, so there's no value to creating new instances when re-using an existing instance would do.

Reply:
Ah, I see